src/ioctls/system.rs

```diff
@@ -45,6 +46,31 @@ impl Kvm {
         Ok(unsafe { Self::from_raw_fd(fd) })
     }
 
+    /// Opens the KVM device at `kvm_path` and returns a `Kvm` object on success.
+    ///
+    /// # Arguments
+    ///
+    /// * `kvm_path`: path to the KVM device. Usually it is `/dev/kvm`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use kvm_ioctls::Kvm;
+    /// use std::ffi::CString;
+    /// let kvm_path = CString::new("/dev/kvm").unwrap();
+    /// let kvm = Kvm::new_with_path(&kvm_path).unwrap();
+    /// ```
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new_with_path<P>(kvm_path: P) -> Result<Self>
+    where
+        P: AsRef<CStr>,
+    {
+        // Open `kvm_path` using `O_CLOEXEC` flag.
+        let fd = Self::open_with_cloexec_at(kvm_path, true)?;
+        // Safe because we verify that the fd is valid in `open_with_cloexec_at` and we own the fd.
```

Comment:
You will need to add the "SAFETY:" suffix before the safety comment to make the CI happy. Otherwise this clippy check will fail: https://rust-lang.github.io/rust-clippy/master/#missing_safety_doc.

Reply:
Done. Thank you very much, now All checks have passed.